Promote statefulset available replicas and docs

yasicar opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. If this is your first time, read our contributor guidelines https://git.k8s.io/community/contributors/guide/pull-requests.md#the-pull-request-submit-process and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
3. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/guide/pull-requests.md#write-release-notes-if-needed
4. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

<!-- markdownlint-disable-next-line MD041 -->
**What this PR does / why we need it:**

**How does this change affect the cardinality of KSM:** *(increases, decreases or does not change cardinality)*

**Which issue(s) this PR fixes:** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*
Fixes #

yasicar commented:
@CatherineF-dev @mrueg Can you please help me with the review

nmn3m commented on docs/metrics/workload/statefulset-metrics.md:
```diff
@@ -18,3 +18,102 @@
+## Common PromQL Queries
+
+### StatefulSet Health Monitoring
+
```
I noticed that you added some PromQL troubleshooting query examples under the metric definition. From my understanding, this file (docs/metrics/workload/statefulset-metrics.md) is meant to document the metrics themselves (type, labels, stability, etc.), not usage or example queries.

yasicar replied:
Hi! Thank you for the feedback. 
I added the PromQL examples because I noticed the Pod metrics documentation [(docs/metrics/workload/pod-metrics.md)](https://github.com/kubernetes/kube-state-metrics/blob/main/docs/metrics/workload/pod-metrics.md#useful-metrics-queries)  includes similar usage examples and queries, and the main README mentions welcoming ["sample usages"](https://github.com/kubernetes/kube-state-metrics/blob/main/docs/README.md) contributions.
I was trying to follow that same pattern for consistency. Please add your thoughts.
I'm happy to adjust based on what works best for the project!

yasicar replied:
@nmn3m

yasicar replied:
@nmn3m Thanks 
This is my first contribution to kube-state-metrics, and I'd really appreciate your help completing this review.

nmn3m commented on internal/store/statefulset.go:
```diff
@@ -82,7 +82,7 @@ func statefulSetMetricFamilies(allowAnnotationsList, allowLabelsList []string) [
 			"The number of available replicas per StatefulSet.",
 			metric.Gauge,
-			basemetrics.ALPHA,
+			basemetrics.STABLE,
```
This metric will need to be in BETA for a quite time then we can promote it to STABLE, 
you can read this as I have the same situation before, https://github.com/kubernetes/kube-state-metrics/pull/2749#issuecomment-3250218117

yasicar replied:
Sure Thanks
I have addressed your comment and updated the entire PR to promote these metrics to BETA instead of STABLE.
https://github.com/kubernetes/kube-state-metrics/pull/2783#:~:text=feat%3A%20promote%20StatefulSet%20experimental%20metrics%20to%20BETA

yasicar commented:
Hi @CatherineF-dev addressed your comments
https://github.com/kubernetes/kube-state-metrics/pull/2783#:~:text=docs%3A%20remove%20PromQL%20queries%20from%20branch 


PR for queries changes  : https://github.com/kubernetes/kube-state-metrics/pull/2821

CatherineF-dev commented:
/lgtm
